composables/use-log-event.ts

```diff
@@ -2,17 +2,55 @@ interface EventParams {
   [key: string]: unknown
 }
 
-export default function useLogEvent(eventName: string, eventParams?: EventParams) {
-  const { instance: crisp } = useScriptCrisp()
+export default function useLogEvent(eventName: string, eventParams: EventParams = {}) {
   try {
     useTrackEvent(eventName, eventParams)
   }
   catch {
     console.error(`Failed to track event: ${eventName}`, eventParams)
   }
+
+  const { proxy } = useScriptMetaPixel()
+  try {
+    const eventNameMapping: { [key: string]: string } = {
+      view_item: 'ViewContent',
+      begin_checkout: 'InitiateCheckout',
+      add_to_cart: 'AddToCart',
+      purchase: 'Purchase',
+      search: 'Search',
+    }
+    if (eventNameMapping[eventName]) {
+      const {
+        transaction_id: paymentId,
+        value,
+        currency,
+        items,
+      } = eventParams
+      const eventID = paymentId ? `${eventName}_${paymentId}` : undefined
+      proxy.fbq('track', eventNameMapping[eventName], {
+        currency,
+        value,
+        order_id: paymentId,
+        content_type: items ? 'product' : undefined,
+        contents: Array.isArray(items)
+          ? items.map(i => ({
+              id: i.id,
+              quantity: i.quantity || 1,
+            }))
+          : undefined,
+        content_ids: Array.isArray(items) ? items.map(i => i.id) : undefined,
+        eventID,
```

Comment:
why mixing snake case and camelCase?

Reply:
https://developers.facebook.com/docs/meta-pixel/implementation/conversion-tracking?locale=zh_HK?
https://developers.facebook.com/docs/marketing-api/conversions-api/deduplicate-pixel-and-server-events/
inside and outside of payload i guess..

Reply:
Moved `eventID` to 3rd param